mailpoet/lib/Util/DataInconsistency/DataInconsistencyRepository.php

```diff
@@ -35,18 +35,6 @@ public function getOrphanedSendingTasksCount(): int {
     return (int)$this->buildOrphanedSendingTasksQuery($builder)->getSingleScalarResult();
   }
 
-  public function getOrphanedScheduledTasksSubscribersCount(): int {
-    $stTable = $this->entityManager->getClassMetadata(ScheduledTaskEntity::class)->getTableName();
-    $stsTable = $this->entityManager->getClassMetadata(ScheduledTaskSubscriberEntity::class)->getTableName();
-    /** @var string $count */
-    $count = $this->entityManager->getConnection()->executeQuery("
-      SELECT count(*) FROM $stsTable sts
-      LEFT JOIN $stTable st ON st.`id` = sts.`task_id`
-      WHERE st.`id` IS NULL
-    ")->fetchOne();
-    return intval($count);
-  }
-
```

Comment:
I gave this a quick try and it seems that splitting it in two queries makes it very fast:

```sql
SELECT task_ids.task_id
FROM (
  SELECT DISTINCT task_id FROM wp_mailpoet_scheduled_task_subscribers
) AS task_ids
LEFT JOIN wp_mailpoet_scheduled_tasks st ON st.id = task_ids.task_id
WHERE st.id IS NULL;

SELECT COUNT(*)
FROM wp_mailpoet_scheduled_task_subscribers sts
WHERE sts.task_id IN (27106,30812,337919,337921);
```

The important piece is the `DISTINCT task_id`. The `wp_mailpoet_scheduled_task_subscribers` table is huge, so JOINing in any way _before_ having just the disctinct task IDs will be ineffective. However, if we first fetch the affected task IDs, we get subsecond query times even on mailpoet.com.

Reply:
@costasovo Here's how to write it with a single query, plus an explaining comment:

```sql
SELECT COUNT(*)
FROM wp_mailpoet_scheduled_task_subscribers sts
WHERE sts.task_id IN (
  -- SELECT ... FROM (subquery) forces the subquery to materialize first.
  -- We're using this twice here to get the reduced data set before JOIN.
  SELECT oprhaned_task_ids.task_id FROM
  (
    SELECT task_ids.task_id
    FROM (
      SELECT DISTINCT task_id FROM wp_mailpoet_scheduled_task_subscribers
    ) AS task_ids
    LEFT JOIN wp_mailpoet_scheduled_tasks st ON st.id = task_ids.task_id
    WHERE st.id IS NULL
    GROUP BY task_ids.task_id
  ) AS oprhaned_task_ids
);
```

This runs in about 0.5s on mailpoet.com.

Reply:
As for `DELETE`, if we want to use it with `LIMIT`, we could probably wrap the whole query with `DELETE FROM (... LIMIT <limit>)`.